src/main/java/org/springframework/data/release/documentation/DocumentationOperations.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.release.documentation;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.fusesource.jansi.Ansi;
+import org.fusesource.jansi.Ansi.Color;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.select.Elements;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Christoph Strobl
+ */
+@Component
+class DocumentationOperations {
+
+	PageStats checkDocumentation(String url) {
+		PageStats stats = new PageStats();
+		stats.links(new LinkChecker().inspect(url));
+		return stats;
+	}
+
+	static class PageStats {
+
+		LinkStats linkStats;
+
+		void prettyPrint() {
+			if (linkStats != null) {
+				linkStats.prettyPrint();
+			}
+		}
+
+		void links(LinkStats linkStats) {
+			this.linkStats = linkStats;
+		}
+	}
+
+	static class LinkStats {
+
+		Map<String, HttpStatus> resultMap = new LinkedHashMap<>(200);
+
+		public HttpStatus computeIfAbsent(String key, Function<? super String, ? extends HttpStatus> mappingFunction) {
+			return resultMap.computeIfAbsent(key, mappingFunction);
+		}
+
+		void prettyPrint() {
+
+			resultMap.entrySet().stream().sorted(Comparator.comparingInt(o -> o.getValue().value())).forEach(entry -> {
+				Ansi ansi = Ansi.ansi();
+				if (entry.getValue().is2xxSuccessful()) {
+					ansi.fg(Color.GREEN);
+				} else if (entry.getValue().is4xxClientError())
+					ansi.fg(Color.RED);
+				else if (entry.getValue().is3xxRedirection()) {
+					ansi.fg(Color.YELLOW);
+				}
+				System.out.println(ansi.a(entry.getValue()).fg(Color.DEFAULT).a(": " + entry.getKey()).toString());
+			});
+		}
+	}
+
+
+	static class LinkChecker {
+
+		LinkStats inspect(String url) throws RuntimeException {
+
+			System.out.println("Collecting links from: " + url);
```

Comment:
We have loggers so we don't need to use sysout. We could have a verbose mode to log everything.

Reply:
ah cool - thanks - will use those then